api.go

```diff
@@ -99,6 +99,12 @@ func (req Spec) Stop() error {
 func NewSpec(stream StreamFunc, cStore CursorStore, consumer Consumer,
 	opts ...StreamOption,
 ) Spec {
+	for _, o := range opts {
+		if o == nil {
+			panic("nil opt passed in")
+		}
+	}
+
```

Comment:
We were weighing up whether to return an error from `Run` instead, to try to avoid panicking where possible.
In the Luno context, that wouldn't cause the deployment to roll back (I think). But we _could_ do an error check when we call it, and panic if err != nil

Reply:
This would help once we start rolling out service testing so then tests will fail. I think panic is the correct approach.

I do feel it's odd that we have nil options being passed in. Validation isn't wrong though so definitely support this change.